### ISSUE TITLE
Bring libtimidity support back

### ISSUE THREAD

**Issue by rsn8887** [OpenTTD/OpenTTD#7636]
This is a reminder to bring libtimidity support back, to enable me to more easily merge the code of my Switch port into this main repo. The Switch port relies on libtimidity for music.

Not having libtimidity causes a plethora of merge errors during my tries to rebase.

See discussion here https://github.com/OpenTTD/OpenTTD/issues/7333, where it was concluded that a port is needed that uses libtimidity to warrant bringing it back. Well thenSwitch port uses it and I am trying to upstream it so there’s that.

And the original commit that removed libtimidity here. https://github.com/OpenTTD/OpenTTD/pull/7326. A simple revert doesn’t seem to work anymore:

**Comment by Milek7** [comment]
It is restored in emscripten PR: https://github.com/OpenTTD/OpenTTD/pull/7510/commits

**Comment by rsn8887** [comment]
Perfect I didn’t see it! Hmm so my rebase should work. Weird. Closing the issue.

**Comment by rsn8887** [comment]
Ah I see it is in a PR that hasn’t been merged yet. Ok that explains it then. I will wait for the PR to be merged.

**Comment by Eddi-z** [comment]
i'm not quite sure how you come to the conclusion that someone **else** should do this for you, instead of you just reverting the commit that removed it?

**Comment by rsn8887** [comment]
I am calling for help here, because tried to revert it myself already, but so much has changed since it was first removed (especially in the configure script or config.libs or whatever it was called) that nothing compiled anymore after I tried to fix the conflicts.

Configure scripts are a constant pain when rebasing/reverting etc. because everything touches them.

**Comment by Milek7** [comment]
```
f04cda523 Revert aa350528: Remove: libtimidity support (NOT timidity support)
c324480c9 Fix: Use sound mixer in libtimidity
673347a7d Codechange: Convert virtual comments to override keywords in libtimidity.h
b00491221 Codechange: Replace NULL with nullptr in libtimidity
```
Cherry-picking those commits from my PR should apply cleanly on master.

**Comment by rsn8887** [comment]
Thanks, that is exactly what I was looking for :)

**Comment by TrueBrain** [comment]
I assume this means we can close this ticket now :) Tnx @Milek7 for helping @rsn8887 out!

**Comment by rsn8887** [comment]
Why was this closed? PR #7510 hasn’t been merged yet.
